enclave-runtime/src/lib.rs

```diff
@@ -291,16 +291,19 @@ pub unsafe extern "C" fn mock_register_enclave_xt(
 	sgx_status_t::SGX_SUCCESS
 }
 
-fn create_extrinsics<PB>(
+fn create_extrinsics<PB, Signer>(
 	genesis_hash: HashFor<PB>,
+	signer: Signer,
 	calls: Vec<OpaqueCall>,
-	nonce: &mut u32,
-) -> Result<Vec<OpaqueExtrinsic>>
+	mut nonce: u32,
+) -> Result<(Vec<OpaqueExtrinsic>, u32)>
```

Comment:
Take the signer as parameter (rather than having the `Ed25519Seal::unseal()?;` side-effect in the function) and the nonce is now an input parameter and returned in the result.